control/tasks.go

```diff
@@ -353,7 +388,7 @@ type Store interface {
 	// parsing error) occurs; otherwise, it returns a slice containing the beacons (which
 	// potentially could be empty when no beacon is found) and no error.
 	// The selections is based on the configured propagation policy for the requested segment type.
-	SegmentsToRegister(ctx context.Context, segType seg.Type) ([]beacon.Beacon, error)
+	SegmentsToRegister(ctx context.Context, segType seg.Type) (beacon.GroupedBeacons, error)
```

Comment:
The comment is no longer up to date. Consider changing to something like:
```
// SegmentsToRegister returns an error if for example connection or
// parsing error occurs; otherwise returns grouped beacons (empty when no beacon is found).
// The selection is based on the configured propagation policy for the requested segment type.
```